tensorflow/lite/c/c_api_types.h

```diff
@@ -30,6 +30,7 @@ extern "C" {
 #ifdef SWIG
 #define TFL_CAPI_EXPORT
 #else
+#ifdef TFL_SHARED_LIBRARY_BUILD
 #if defined(_WIN32)
 #ifdef TFL_COMPILE_LIBRARY
```

Comment:
Then show about the following?
```
#if defined(_WIN32)
#if defined(TFL_SHARED_LIBRARY_BUILD) ||  defined(TFL_COMPILE_LIBRARY)
```

I'd like to reduce the depth of the conditions.

Reply:
Probably. But there is also branch for not Windows depended on TFL_COMPILE_LIBRARY.

Reply:
sorry, I don't get it. Could you elaborate?
The definition TFL_COMPILE_LIBRARY is only used under  _WIN32 in the original code.

Reply:
Sorry. My mistake. 

But I just realize. You can join checks unfortunately. There are 3 outcome needed. 

```
If build_shared
  If build_library
    Tfl_capi=dllexport
  Else
    Tfl_capi=dllimport
Else
  Tfl_capi=
```

Probably whole this block could be moved under `ifdef WIN32` but for not win32 checks TFL_BUILD_SHARED_LIBRARY still needed.

Reply:
I see. One issue is that TFL_SHARED_LIBRARY_BUILD will be defined only with CMake so it will break Bazel builds.

How about defining TFL_STATIC_LIBRARY_BUILD instead?
Then the logic will be
```
#if defined(_WIN32)
#if defined(TFL_STATIC_LIBRARY_BUILD)
#define TFL_CAPI_EXPORT
#elif defined(TFL_COMPILE_LIBRARY)
#define TFL_CAPI_EXPORT __declspec(dllexport)
#else
#define TFL_CAPI_EXPORT __declspec(dllimport)
#endif  // TFL_STATIC_LIBRARY_BUILD
```

Reply:
For me it's ok. But I know nothing about Bazel and do not know where fix it?

Reply:
In Bazel, TFL_STATIC_LIBRARY_BUILD won't be defined.
So the logic will be identical to the original logic which means no regression.

Reply:
Ok. I will fix it later today.

Reply:
ping?

Reply:
Fixed. Sorry, suddenly was AWK